client-core/src/service/wallet_service.rs

```diff
@@ -15,16 +14,65 @@ use client_common::{
 const KEYSPACE: &str = "core_wallet";
 
 /// Wallet meta data
-#[derive(Debug, Encode, Decode)]
+#[derive(Debug)]
 pub struct Wallet {
     /// view key to decrypt enclave transactions
     pub view_key: PublicKey,
     /// public keys to construct transfer addresses
-    pub public_keys: BTreeSet<PublicKey>,
+    pub public_keys: IndexSet<PublicKey>,
     /// public keys of staking addresses
-    pub staking_keys: BTreeSet<PublicKey>,
+    pub staking_keys: IndexSet<PublicKey>,
     /// root hashes of multi-sig transfer addresses
-    pub root_hashes: BTreeSet<H256>,
+    pub root_hashes: IndexSet<H256>,
+}
+
+impl Encode for Wallet {
```

Comment:
Can you write some tests for testing `Encode` and `Decode` implementations?

Reply:
It is covered in `check_flow` test case, most of the service APIs involves encoding/decoding of the `Wallet`.

Reply:
Ok